### PR TITLE
[Pools] Ability for mempool to remove transactions that fail block validity

### PR DESCRIPTION
**Pull request by blondfrogs**
When **getblocktemplate** is called it takes in transactions from the mempool to build a block. If one of those transactions ends up being invalid the **getblocktemplate** rpc call will fail. This can cause pools to stop producing valid block data for their miners to solve. 

This code change makes it so that if **getblocktemplate** fails to produce a valid block because of transactions in the block. The invalid transactions will be removed from the mempool so that the next time **getblocktemplate** is called it won't fail over and over

### CODE REVIEW

File: src/consensus/validation.h [hunk 1]
```diff
@@ -76,6 +80,17 @@ class CValidationState {
     }
     virtual unsigned char GetRejectCode() const { return chRejectCode; }
     virtual std::string GetRejectReason() const { return strRejectReason; }
+
+    virtual bool IsInvalidTx() const {
+        return txInvalidTx.IsNull() == false;
+    }
+    virtual CTransaction GetInvalidTx() const {
```
Comment (johnhanlon86) [inline src/consensus/validation.h:87]: Is there any reason why these two functions are grouped together or should they be separate?
```suggestion
    }
    
    virtual CTransaction GetInvalidTx() const {
```

Comment (blondfrogs) [reply src/consensus/validation.h:87]: @johnhanlon86 what do you mean?

Comment (johnhanlon86) [reply src/consensus/validation.h:87]: @blondfrogs Just checking if you want whitespace/line-breaks between functions... more of a styling question than a technical one :)

Comment (blondfrogs) [reply src/consensus/validation.h:87]: Oh doesn't matter atm

**Review by TheTrunk — APPROVED** [review]
tACK

**Review by Cabecinha84 — APPROVED** [review]
Ack

**Pull request merged**